### ISSUE TITLE
[SR-11593] REPL incorrectly reports URL as nil

### ISSUE THREAD

**Issue by mattneub** [swiftlang/llvm-project#4477]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-11593      |
   |Radar             | None         |
   |Original Reporter | @mattneub      |
   |Type              | Bug    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 16         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | jingham@apple.com (JIRA)      |
   |Priority          | Medium      |

   

   md5: 52c0d59c4f387bd8ac5d0f53094bdb03

  </details>

**is blocked by**:  
* [SR-11546](https://bugs.swift.org/browse/SR-11546) LLDB gets Swift Foundation overlay optionals wrong  

**duplicates**:  
* [SR-13103](https://bugs.swift.org/browse/SR-13103) lldb cannot evaluate optional date objects  

**is duplicated by**:  
* [SR-12444](https://bugs.swift.org/browse/SR-12444) Xcode Debugger and lldb fail to display Foundation Types  
* [SR-13446](https://bugs.swift.org/browse/SR-13446) Optional<IndexPath>.none != nil  

**relates to**:  
* [SR-4091](https://bugs.swift.org/browse/SR-4091) Incorrect result shown in REPL for 'UInt8?'  
* [SR-11546](https://bugs.swift.org/browse/SR-11546) LLDB gets Swift Foundation overlay optionals wrong  

**Issue Description:**

Here's a curious little conversation with the command line that came up on Stack Overflow lately:

``` java
themini:~ mattneubelcap$ swift
Welcome to Apple Swift version 5.1 (swiftlang-1100.0.270.13 clang-1100.0.33.7).
Type :help for assistance.
  1> import Foundation
  2> let url = URL(string:"https://bugs.swift.org")
url: URL? = nil
  3> print(url)
Optional(https://bugs.swift.org)
  4> url == nil
$R0: Bool = false
  5> url
$R1: URL? = nil
```

I have not tried to determine what other types behave this way, but I suspect is has to do with the Foundation overlay. This is interesting, if so, because I recently reported another bug involving LLDB incorrectly reporting nil for Optional Foundation overlay types (<https://bugs.swift.org/browse/SR-11546).>

**Comment by belkadan** [comment]
Indeed, very possibly the same.

**Comment by kastiglione** [comment]
MacDavePro (JIRA User) for the debugger, a workaround is to use \`po\`
